### ISSUE TITLE
<Miss> on server: Cannot set property 'hasMissComponent' of undefined

### ISSUE THREAD

**Issue by stnwk** [ReactTraining/react-router#3862]
When using the Miss Component as follows in a ssr-environment the following error appears on the server-side:
```js
          <Match exactly pattern="/" component={Hello} />
          <Match pattern="/news" component={News} />
          <Miss component={() => (<span>missed.</span>)} />
```

SSR is set up according to the v4 docs.

**Error:**

```
TypeError: Cannot set property 'hasMissComponent' of undefined
    at Object.registerMissPresence (/Users/Stefan/Documents/www/projects/starterkit/build/server.js:109468:45)
    at new Miss (/Users/Stefan/Documents/www/projects/starterkit/build/server.js:107228:30)
```

**Comment by stnwk** [comment]
This has something to do when using apollo in server-side-rendering, especially when using it's getDataFromTree() method.

Any idea on how to set up React Router correctly with apollo?

**Comment by ryanflorence** [comment]
server rendering "Miss" is still a bit of an open question.

**Comment by stnwk** [comment]
Resolved by updating react-apollo to 0.5.5, but now leads to #3877 
Closing, thanks.
